src/hooks/useDnd.ts

```diff
@@ -0,0 +1,74 @@
+import { useState } from 'react';
```

Comment:
I think it's better to rename the hook to more special, because it's using specially for folders and applets dnd.